testingutils/src/main/java/com/github/awsjavakit/testingutils/aws/FakeSqsClient.java

```diff
@@ -0,0 +1,65 @@
+package com.github.awsjavakit.testingutils.aws;
+
+import static java.util.Objects.isNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchRequestEntry;
+import software.amazon.awssdk.services.sqs.model.SendMessageBatchResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
+
+public class FakeSqsClient implements SqsClient {
```

Comment:
Vague thoughts here…is this an honest attempt to replicate SQS functionality? If so, I guess there are more things around config and behaviours that need to be thought about than I care to think about :D

Reply:
I am not really trying to replicate SQS functionality. Most of the time, the subscribers do not interact with the client. It is only the producers that use the SQS client. If we would like to test the subscriber and producer simultaneously, then we would have to add more functionality to the fake client.